### PR TITLE
Feature/#397 Setup now does not install helm if helm v3 already installed globally

### PR DESCRIPTION
**Pull request by wavell**
# Setup now does not install helm if helm v3 already installed globally

## Description
Setup now does not install helm if helm v3 already installed globally

## Issues:
Refs: #397

## How has this been tested:
 - [ ] Covered by existing integration testing
 - [x] Added integration testing to cover
 - [ ] Verified all A/C passes
     * [ ] develop
     * [ ] master
     * [ ] tag/other branch
 - [ ] Test environment
    * [ ] Shared Packet K8s cluster
    * [ ] New Packet K8s cluster
    * [ ] Kind cluster
 - [ ] Have not tested

## Types of changes:
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Documentation update

## Checklist:
**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. rescue block

**Issue**
- [ ] Tasks in issue are checked off

### CODE REVIEW

File: spec/utils/cnf_manager_spec.cr [hunk 1]
```diff
@@ -32,7 +32,8 @@ describe "SampleUtils" do
 
     current_dir = FileUtils.pwd 
     LOGGING.info current_dir
-    helm = "#{current_dir}/#{TOOLS_DIR}/helm/linux-amd64/helm"
+    #helm = "#{current_dir}/#{TOOLS_DIR}/helm/linux-amd64/helm"
+    helm = CNFSingleton.helm
```
Comment (taylor) [inline spec/utils/cnf_manager_spec.cr:36]: @wavell where is CNFSingleton in the codebase... Is that magic Crystal lang stuff?  I don't see the usage in the Crystal Singleton module.  Is this just a naming convention for the variable including the "type" (singleton)?

File: src/tasks/helmenv_setup.cr [hunk 2]
```diff
@@ -6,30 +6,34 @@ require "./utils/utils.cr"
...
+    VERBOSE_LOGGING.debug current_dir if check_verbose(args)
+    unless Dir.exists?("#{current_dir}/#{TOOLS_DIR}/helm")
+      begin
+        VERBOSE_LOGGING.debug "pwd? : #{current_dir}" if check_verbose(args)
+        VERBOSE_LOGGING.debug "toolsdir : #{TOOLS_DIR}" if check_verbose(args)
+        VERBOSE_LOGGING.debug "full path?: #{current_dir.to_s}/#{TOOLS_DIR}/helm" if check_verbose(args)
+        FileUtils.mkdir_p("#{current_dir}/#{TOOLS_DIR}/helm") 
+        wget = `wget https://get.helm.sh/helm-v3.1.1-linux-amd64.tar.gz -O #{current_dir}/#{TOOLS_DIR}/helm/helm-v3.1.1-linux-amd64.tar.gz`
+        VERBOSE_LOGGING.debug wget if check_verbose(args)
+        tar = `cd #{current_dir}/#{TOOLS_DIR}/helm; tar -xvf #{current_dir}/#{TOOLS_DIR}/helm/helm-v3.1.1-linux-amd64.tar.gz`
```
Comment (taylor) [inline src/tasks/helmenv_setup.cr:23]: The cd command could be removed if the `-C` option is used for tar.  Eg.

`tar -xvfC #{current_dir}/#{TOOLS_DIR}/helm/helm-v3.1.1-linux-amd64.tar.gz #{current_dir}/#{TOOLS_DIR}/helm`

Comment (taylor) [inline src/tasks/helmenv_setup.cr:23]: Suggestion: make the helm tar file or at least the version a variable which is used throughout the helm_local_install task.

File: src/tasks/helmenv_setup.cr [hunk 3]
```diff
@@ -6,30 +6,34 @@ require "./utils/utils.cr"
...
+        stable_repo = `#{helm} repo add stable https://kubernetes-charts.storage.googleapis.com`
+        VERBOSE_LOGGING.debug stable_repo if check_verbose(args)
 
-      #TODO grep for version.BuildInfo{Version:"v3.1.1", GitCommit:"afe70585407b420d0097d07b21c47dc511525ac8", GitTreeState:"clean", GoVersion:"go1.13.8"} 
-    ensure
-      cd = `cd #{current_dir}`
-      VERBOSE_LOGGING.debug cd if check_verbose(args)
+        #TODO grep for version.BuildInfo{Version:"v3.1.1", GitCommit:"afe70585407b420d0097d07b21c47dc511525ac8", GitTreeState:"clean", GoVersion:"go1.13.8"} 
+      ensure
+        cd = `cd #{current_dir}`
```
Comment (taylor) [inline src/tasks/helmenv_setup.cr:34]: Suggestion: use the Crystal lang builtin cd command, Dir.cd.  See https://crystal-lang.org/api/0.21.0/Dir.html#cd%28path%2C%26block%29-class-method

**Review by taylor — APPROVED** [review]
I added some comments, questions, and a few suggestions.  Overall though lgtm and no blockers.

File: spec/utils/cnf_manager_spec.cr (hunk 1 above)
Comment (wavell) [reply spec/utils/cnf_manager_spec.cr:36]: src/tasks/constants.cr

File: src/tasks/helmenv_setup.cr (hunk 3 above)
Comment (taylor) [reply src/tasks/helmenv_setup.cr:35]: suggestion added to ticket#410

File: src/tasks/helmenv_setup.cr (hunk 2 above)
Comment (taylor) [reply src/tasks/helmenv_setup.cr:23]: Suggestion added to ticket #409

**Comment by wavell** [comment]
Someone needs to accept this pull request!

**Comment by nupejosh** [comment]
@wavell doing AC now

**Pull request merged**